src/mlpack/methods/ann/layer/celu.hpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file celu.hpp
+ * @author Gaurav Singh
+ *
+ * Definition of the CELU activation function as described by Jonathan T. Barron.
+ *
+ * For more information, read the following paper.
+ *
+ * @code
+ * @article{
+ *   author  = {Jonathan T. Barron},
+ *   title   = {Continuously Differentiable Exponential Linear Units},
+ *   year    = {2017}
```

Comment:
Hey @gaurav-singh1998, I think @codeboy5, added papers for most of the layers in  #2267. It would be great if you could add it here as well if possible. Thanks.